primitives/src/lib.rs

```diff
@@ -6,30 +6,27 @@ use scale::{Decode, Encode, MaxEncodedLen};
 use scale_info::TypeInfo;
 pub use v0::error;
 
-#[cfg(feature = "cross-chain")]
-pub mod cross_chain;
-pub mod storage_keys;
-
 /// An opaque 32-byte cryptographic identifier.
 #[derive(Encode, Decode, Debug, MaxEncodedLen, Eq, PartialEq)]
 #[cfg_attr(feature = "std", derive(TypeInfo))]
 pub struct AccountId(pub [u8; 32]);
 
 /// Identifier for the class of asset.
 pub type AssetId = u32;
+pub const FUNGIBLES: u8 = 150;
 
-#[cfg(feature = "nfts")]
-pub mod nfts {
-	use bounded_collections::ConstU32;
-
-	/// Id used for identifying non-fungible collections.
-	pub type CollectionId = u32;
-	/// Id used for identifying non-fungible items.
-	pub type ItemId = u32;
-	/// The maximum length of an attribute key.
-	pub type KeyLimit = ConstU32<64>;
-	/// The maximum approvals an item could have.
-	pub type ApprovalsLimit = ConstU32<20>;
+#[cfg(feature = "fungibles")]
+pub mod constants {
+	/// Module index.
+	pub const FUNGIBLES: u8 = 150;
+	pub mod fungibles {
+		pub const TOTAL_SUPPLY: u8 = 0;
```

Comment:
Is this not overly complex? Does using an index attribute on the enum in the runtime and hardcoding the u8 value in the pop-api function not achieve the same thing without the additional 'coupling'.

Seems additional maintenance and tons of consts just to guarantee that the same index is used, when this would probably be covered by some test to ensure that functionality works as expected.

I figure its far simpler to define the index on the enum in the runtime and then just specify the same value in the relevant function within pop_api crate.

Reply:
It depends what we want. If we think this overcomplicates things too much we can definitely do what we had before. I thought the 'coupling' is something that we wanted and therefore implemented this to show.

Reply:
Removed the constants from `primitives`.  Curious to hear whether we should use them in pop api. Previously this was desired as it improves readability but does add a lot of constants.

Reply:
No clear best approach sadly. In some circumstances the constants seem better, but its just a pointer to a special number used in a single place, then debatable on how much value it brings.

Perhaps we defer making this decision until everything comes together?

Reply:
Sure thing. Created #124